Added the xWeakHostSend and xWeakHostReceive resource - Fixes #266

Warryz opened the pull request:
**Pull Request (PR) description**
Hello everyone,

I'm excited to create my 1st pull request to a github repositoy. I created two new resources for changing the weak host send and weak host receive setting on a network adapter. It's a pretty useful setting for multihomed hosts. 

If there are any issues or questions feel free to ask. ;) 

**This Pull Request (PR) fixes the following issues:**
This is not really a fix, but an improvement. It adds two resources as mentioned in the issue #266.

**Task list:**
- [x] Change details added to Unreleased section of CHANGELOG.md?
- [x] Added/updated documentation, comment-based help and descriptions in .schema.mof files where appropriate?
- [x] Examples appropriately updated?
- [x] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [x] [Unit and (optional) Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xnetworking/289)
<!-- Reviewable:end -->

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/PowerShell/xNetworking?pullRequest=289) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/PowerShell/xNetworking?pullRequest=289) before we can accept your contribution.<br/><br/>:x: Warryz [sign now](https://cla.opensource.microsoft.com/PowerShell/xNetworking?pullRequest=289)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/PowerShell/xNetworking?pullRequest=289) it.</sub>

PlagueHO commented:
Really amazing job @Warryz - only some really trivial style nitpicks. Your work on this is very much appreciated! Thank you for submitting this. 

---

Reviewed 2 of 8 files at r1, 12 of 18 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 23 unresolved discussions.

---

*[CHANGELOG.md, line 5 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-taztz6bMPZcAxdlsX:-L-tazu-w7A-wQZ2q-KV:bes34rw) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/CHANGELOG.md#L5)):*
> ```Markdown
> ## Unreleased
> 
> - xWeakHostSend
> ```

For consistency with other entries, can you change this to `MSFT_xWeakHostSend:` including `:`.

---

*[CHANGELOG.md, line 7 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tb6Yu7PScRxV1ToWi:-L-tb6Yu7PScRxV1ToWj:b2vurab) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/CHANGELOG.md#L7)):*
> ```Markdown
> - xWeakHostSend
>   - Created the new Weak Host Send resource.
> - xWeakHostReceive
> ```

For consistency with other entries, can you change this to `MSFT_xWeakHostReceive:` including `:`.

---

*[Modules/xNetworking/DSCResources/MSFT_xWeakHostReceive/MSFT_xWeakHostReceive.psm1, line 126 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tbSCxZ8jduapT9_cn:-L-tbSCyasBhEyG8vWus:b-xzfrpv) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/DSCResources/MSFT_xWeakHostReceive/MSFT_xWeakHostReceive.psm1#L126)):*
> ```PowerShell
>                 -f $InterfaceAlias, $AddressFamily, $State `
>         ) -join '' )
> ```

Nitpick: Can you remove this blank line?

---

*[Modules/xNetworking/DSCResources/MSFT_xWeakHostSend/MSFT_xWeakHostSend.psm1, line 126 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tcFIeTOTa_XnWsK2Q:-L-tcFIeTOTa_XnWsK2R:b2f7koz) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/DSCResources/MSFT_xWeakHostSend/MSFT_xWeakHostSend.psm1#L126)):*
> ```PowerShell
>                 -f $InterfaceAlias, $AddressFamily, $State `
>         ) -join '' )
> ```

Can you remove this blank line?

---

*[Modules/xNetworking/Examples/Resources/xWeakHostReceive/1-xConfigureWeakHostReceive.ps1, line 22 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tcds_AK6R42FNC1_u:-L-tcds_AK6R42FNC1_v:b-s64pxl) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/Examples/Resources/xWeakHostReceive/1-xConfigureWeakHostReceive.ps1#L22)):*
> ```PowerShell
>             State          = 'Disabled'
>             InterfaceAlias = 'Ethernet'
>             AddressFamily = 'IPv4'
> ```

Nitpic: Align = for AddressFamily under = for other parameters (sorry for being so obsessive :grin:). VS Code has a setting in _Format Document_ that will do this automatically if it helps.

---

*[Modules/xNetworking/Examples/Resources/xWeakHostSend/1-xConfigureWeakHostSend.ps1, line 22 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tcwcC3ydA9PKQTlxn:-L-tcwcC3ydA9PKQTlxo:b-s64pxl) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/Examples/Resources/xWeakHostSend/1-xConfigureWeakHostSend.ps1#L22)):*
> ```PowerShell
>             State          = 'Disabled'
>             InterfaceAlias = 'Ethernet'
>             AddressFamily = 'IPv4'
> ```

Nitpic: Align = for AddressFamily under = for other parameters (sorry for being so obsessive :grin:). VS Code has a setting in _Format Document_ that will do this automatically if it helps.

---

*[Tests/Integration/MSFT_xWeakHostReceive.Integration.Tests.ps1, line 37 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-td3bdKKzIjgFmSdx1:-L-td3bdKKzIjgFmSdx2:b-z8wtp6) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Integration/MSFT_xWeakHostReceive.Integration.Tests.ps1#L37)):*
> ```PowerShell
>             {
>                 & "$($script:DSCResourceName)_Config" -OutputPath $TestDrive
>                 Start-DscConfiguration -Path $TestDrive -ComputerName localhost -Wait -Verbose -Force
> ```

We've started adding the `-ErrorAction Stop` parameter to the `Start-DscConfiguration` calls in integration tests to ensure that it always fails on errors - so can you add this parameter?

---

*[Tests/Integration/MSFT_xWeakHostSend.Integration.Tests.ps1, line 37 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdNq5-z9rcyIInBVT:-L-tdNq5-z9rcyIInBVU:b-z8wtp6) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Integration/MSFT_xWeakHostSend.Integration.Tests.ps1#L37)):*
> ```PowerShell
>             {
>                 & "$($script:DSCResourceName)_Config" -OutputPath $TestDrive
>                 Start-DscConfiguration -Path $TestDrive -ComputerName localhost -Wait -Verbose -Force
> ```

We've started adding the `-ErrorAction Stop` parameter to the `Start-DscConfiguration` calls in integration tests to ensure that it always fails on errors - so can you add this parameter?

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdXljaVmDhhfZABWq:-L-tdXljaVmDhhfZABWr:bbb1swi) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L1)):*
> ```PowerShell
> <#
> ```

Can you delete the comment as we generally remove this header out of the template once it is completed (I'm still going through removing old ones too).

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 39 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdqfkPsDEkAAAGv5V:-L-tdqfkPsDEkAAAGv5W:busoxb7) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L39)):*
> ```PowerShell
> 
> function Invoke-TestSetup {
>     # TODO: Optional init code goes here...
> ```

Can you delete this comment?

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 45 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdt4aV1ke89hJgFJl:-L-tdt4aV1ke89hJgFJm:baypy92) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L45)):*
> ```PowerShell
>     Restore-TestEnvironment -TestEnvironment $TestEnvironment
> 
>     # TODO: Other Optional Cleanup Code Goes Here...
> ```

Can you delete this comment and the blank line above?

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 59 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjmgSFzTHqIsWf921:-L-tjmgSFzTHqIsWf922:bvs3jvs) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L59)):*
> ```PowerShell
>         }
> 
>         $testNetIPInterfaceEnabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$testNetIPWeakHostReceiveEnabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 65 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjqkNHlxizQm9pChl:-L-tjqkOtFJaRSxjAYUA:blf0ouj) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L65)):*
> ```PowerShell
>         }
> 
>         $testNetIPInterfaceDisabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$testNetIPWeakHostReceiveDisabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 71 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjsN4Dfk5cI-Lu7bw:-L-tjsN4Dfk5cI-Lu7bx:b-a064k0) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L71)):*
> ```PowerShell
>         }
> 
>         $mockNetIPInterfaceEnabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$mockNetIPWeakHostReceiveEnabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 77 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjvWzS3Hs3t1rq_p3:-L-tjvWzS3Hs3t1rq_p4:b4mpmtf) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L77)):*
> ```PowerShell
>         }
> 
>         $mockNetIPInterfaceDisabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$mockNetIPWeakHostReceiveDisabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 265 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tikfuU0d_jxPzJmkg:-L-tikfuU0d_jxPzJmkh:b-ty5tum) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L265)):*
> ```PowerShell
> {
>     #region FOOTER
>     Restore-TestEnvironment -TestEnvironment $TestEnvironment
> ```

I don't think you need this line - because this command is called inside `Invoke-TestCleanup` - so you could remove the entire #region FOOTER

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 18 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-te8xc6spmzIUnjkPN:-L-te8xdJdjx5aIclLYD:busoxb7) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L18)):*
> ```PowerShell
> 
> function Invoke-TestSetup {
>     # TODO: Optional init code goes here...
> ```

Can you delete this comment?

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 24 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-te6CqLnuLNTMJ-Scr:-L-te6CqLnuLNTMJ-Scs:baypy92) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L24)):*
> ```PowerShell
>     Restore-TestEnvironment -TestEnvironment $TestEnvironment
> 
>     # TODO: Other Optional Cleanup Code Goes Here...
> ```

Can you delete this comment and the blank line above?

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 38 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjFE6HEVrjGdiSkN_:-L-tjFE6HEVrjGdiSkNa:b-ydeyl7) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L38)):*
> ```PowerShell
>         }
> 
>         $testNetIPInterfaceEnabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$testNetIPWeakHostSendEnabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 44 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjVpThPOK5nrEA7c0:-L-tjVpThPOK5nrEA7c1:bukcdim) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L44)):*
> ```PowerShell
>         }
> 
>         $testNetIPInterfaceDisabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$testNetIPWeakHostSendDisabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 50 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjXqoLwNO-i0-sqfT:-L-tjXqoLwNO-i0-sqfU:b9msi8t) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L50)):*
> ```PowerShell
>         }
> 
>         $mockNetIPInterfaceEnabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$mockNetIPWeakHostSendEnabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 56 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjak5cLG6Nx3-O-3q:-L-tjak5cLG6Nx3-O-3r:b-pvh85m) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L56)):*
> ```PowerShell
>         }
> 
>         $mockNetIPInterfaceDisabled = [PSObject]@{
> ```

Nitpick: Can you change this variable to `$mockNetIPWeakHostSendDisabled` - a search replace should sort this fairly easily.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 244 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tiuaSX4To-dQeZ-lp:-L-tiuaSX4To-dQeZ-lq:b-ty5tum) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L244)):*
> ```PowerShell
> {
>     #region FOOTER
>     Restore-TestEnvironment -TestEnvironment $TestEnvironment
> ```

I don't think you need this line - because this command is called inside `Invoke-TestCleanup` - so you could remove the entire #region FOOTER

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289#-:-L-tkB8eC_e8dXY3P2eU:b3w41i5)*
<!-- Sent from Reviewable.io -->

Warryz commented:
Thank you for your review @PlagueHO . I'm going to resolve these style issue in another commit. 👍

Warryz commented:
Review status: all files reviewed at latest revision, 23 unresolved discussions.

---

*[CHANGELOG.md, line 5 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-taztz6bMPZcAxdlsX:-L-uO1-Z0iNkNucO_Bq8:b-jfm05q) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/CHANGELOG.md#L5)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

For consistency with other entries, can you change this to `MSFT_xWeakHostSend:` including `:`.
</blockquote></details>

Changed it.

---

*[CHANGELOG.md, line 7 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tb6Yu7PScRxV1ToWi:-L-uT-1WvNi37J0XF2PD:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/CHANGELOG.md#L7)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

For consistency with other entries, can you change this to `MSFT_xWeakHostReceive:` including `:`.
</blockquote></details>

Done.

---

*[Modules/xNetworking/DSCResources/MSFT_xWeakHostReceive/MSFT_xWeakHostReceive.psm1, line 126 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tbSCxZ8jduapT9_cn:-L-uTQd6gd08eL7nzxF5:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/DSCResources/MSFT_xWeakHostReceive/MSFT_xWeakHostReceive.psm1#L126)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you remove this blank line?
</blockquote></details>

Done.

---

*[Modules/xNetworking/DSCResources/MSFT_xWeakHostSend/MSFT_xWeakHostSend.psm1, line 126 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tcFIeTOTa_XnWsK2Q:-L-uTd8E9zL6nx0LGKCR:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/DSCResources/MSFT_xWeakHostSend/MSFT_xWeakHostSend.psm1#L126)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you remove this blank line?
</blockquote></details>

Done.

---

*[Modules/xNetworking/Examples/Resources/xWeakHostReceive/1-xConfigureWeakHostReceive.ps1, line 22 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tcds_AK6R42FNC1_u:-L-uTohlt5EcVj1JIrfv:b-cqfzjm) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/Examples/Resources/xWeakHostReceive/1-xConfigureWeakHostReceive.ps1#L22)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpic: Align = for AddressFamily under = for other parameters (sorry for being so obsessive :grin:). VS Code has a setting in _Format Document_ that will do this automatically if it helps.
</blockquote></details>

I used notepad++ before and read in one of your comments about VS code. I tried it and it's fun :smile: Resolved anyway :+1: 

---

*[Modules/xNetworking/Examples/Resources/xWeakHostSend/1-xConfigureWeakHostSend.ps1, line 22 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tcwcC3ydA9PKQTlxn:-L-uU6titVqrUzOtmJgI:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Modules/xNetworking/Examples/Resources/xWeakHostSend/1-xConfigureWeakHostSend.ps1#L22)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpic: Align = for AddressFamily under = for other parameters (sorry for being so obsessive :grin:). VS Code has a setting in _Format Document_ that will do this automatically if it helps.
</blockquote></details>

Done.

---

*[Tests/Integration/MSFT_xWeakHostReceive.Integration.Tests.ps1, line 37 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-td3bdKKzIjgFmSdx1:-L-uUPPWVMaFK60widK6:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Integration/MSFT_xWeakHostReceive.Integration.Tests.ps1#L37)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

We've started adding the `-ErrorAction Stop` parameter to the `Start-DscConfiguration` calls in integration tests to ensure that it always fails on errors - so can you add this parameter?
</blockquote></details>

Done.

---

*[Tests/Integration/MSFT_xWeakHostSend.Integration.Tests.ps1, line 37 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdNq5-z9rcyIInBVT:-L-uVuPz4kjbSCkPrM2e:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Integration/MSFT_xWeakHostSend.Integration.Tests.ps1#L37)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

We've started adding the `-ErrorAction Stop` parameter to the `Start-DscConfiguration` calls in integration tests to ensure that it always fails on errors - so can you add this parameter?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289)*
<!-- Sent from Reviewable.io -->

Warryz commented:
Review status: all files reviewed at latest revision, 23 unresolved discussions.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdXljaVmDhhfZABWq:-L-undX08XsmTcplDWjW:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L1)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you delete the comment as we generally remove this header out of the template once it is completed (I'm still going through removing old ones too).
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 39 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdqfkPsDEkAAAGv5V:-L-uo7XXawhUfm1s71e2:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L39)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you delete this comment?
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 45 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tdt4aV1ke89hJgFJl:-L-uoDtoLL5Oo4GjAFiz:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L45)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you delete this comment and the blank line above?
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 59 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjmgSFzTHqIsWf921:-L-uoMXMutcPs7VEsar7:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L59)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$testNetIPWeakHostReceiveEnabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 65 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjqkNHlxizQm9pChl:-L-uoT6eAtMDXIba_Xb0:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L65)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$testNetIPWeakHostReceiveDisabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 71 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjsN4Dfk5cI-Lu7bw:-L-uoguCad599pF5GX0m:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L71)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$mockNetIPWeakHostReceiveEnabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 77 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjvWzS3Hs3t1rq_p3:-L-uoo6LLz9S6VeIEdPW:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L77)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$mockNetIPWeakHostReceiveDisabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1, line 265 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tikfuU0d_jxPzJmkg:-L-up4bypUMW303YNCfM:blyl3ub) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostReceive.Tests.ps1#L265)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I don't think you need this line - because this command is called inside `Invoke-TestCleanup` - so you could remove the entire #region FOOTER
</blockquote></details>

You're right, removed it.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 18 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-te8xc6spmzIUnjkPN:-L-upM0Z6IJWWXLgD6pp:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L18)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you delete this comment?
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 24 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-te6CqLnuLNTMJ-Scr:-L-upP2rBTMs3EFzJsET:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L24)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Can you delete this comment and the blank line above?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289)*
<!-- Sent from Reviewable.io -->

Warryz commented:
Review status: all files reviewed at latest revision, 23 unresolved discussions.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 44 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjVpThPOK5nrEA7c0:-L-urf0-V2hT9cjtlTMY:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L44)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$testNetIPWeakHostSendDisabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289)*
<!-- Sent from Reviewable.io -->

Warryz commented:
Review status: 9 of 18 files reviewed at latest revision, 23 unresolved discussions.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 38 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjFE6HEVrjGdiSkN_:-L-wUkAMdLuOQR3DYRpI:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L38)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$testNetIPWeakHostSendEnabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 50 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjXqoLwNO-i0-sqfT:-L-wUl0GWZZJdhIqhJkT:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L50)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$mockNetIPWeakHostSendEnabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 56 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tjak5cLG6Nx3-O-3q:-L-wUlZMS6r4HPGrjy1p:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L56)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Nitpick: Can you change this variable to `$mockNetIPWeakHostSendDisabled` - a search replace should sort this fairly easily.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 244 at r3](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-tiuaSX4To-dQeZ-lp:-L-wUlrUxUP-QPwp3PWZ:b-896fix) ([raw file](https://github.com/powershell/xnetworking/blob/150a7c727e3b7126356e6281c0596b3c49cb84d9/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L244)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

I don't think you need this line - because this command is called inside `Invoke-TestCleanup` - so you could remove the entire #region FOOTER
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289)*
<!-- Sent from Reviewable.io -->

Warryz commented:
Hey @PlagueHO, I changed the style issues, it should be fine now 😄

PlagueHO commented:
Really great job @Warryz - everything sorted except I think the variable renaming in `MSFT_xWeakHostSend.Tests.ps1` may have gone wrong. So if you can fix that I'll get this merged.

Again, fantastic work and these will make a great addition to the resource kit.

---

Reviewed 9 of 9 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 35 at r4](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-wqp8t5lQVRXACyfC1:-L-wqp8t5lQVRXACyfC2:b-xjh6qv) ([raw file](https://github.com/powershell/xnetworking/blob/df7df734d536211cdc57c88b71d9cdb6f9ad707c/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L35)):*
> ```PowerShell
>         }
> 
>         $mockNetIPWeakHostReceiveDisabled = [PSObject]@{
> ```

It looks like this line is incorrect: Should be `$testNetIPWeakHostSendEnabled`

---

*[Tests/Unit/MSFT_xWeakHostSend.Tests.ps1, line 48 at r4](https://reviewable.io:443/reviews/powershell/xnetworking/289#-L-wqBluSKkoskmypIKm:-L-wqBluSKkoskmypIKn:b-t5ok6x) ([raw file](https://github.com/powershell/xnetworking/blob/df7df734d536211cdc57c88b71d9cdb6f9ad707c/Tests/Unit/MSFT_xWeakHostSend.Tests.ps1#L48)):*
> ```PowerShell
> 
>         $mockNetIPWeakHostSendEnabled = [PSObject]@{
>             WeakHostSend = $mockNetIPWeakHostReceiveDisabled.State
> ```

It looks like these 3 lines aren't correct - should be:
```powershell
WeakHostSend = $testNetIPWeakHostSendDisabled.State
InterfaceAlias  = $testNetIPWeakHostSendDisabled.Name
AddressFamily   = $testNetIPWeakHostSendDisabled.AddressFamily
```

I think that has also affected the other lines in this file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289#-:-L-wr6dg5gw1zSGTX6NG:b8emdno)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@Warryz - I merged another PR which has caused a minor merge conflict in the CHANGELOG.MD - so you'll want to rebase your branch against PowerShell/xNetworking:Dev as well.

Warryz commented:
hey @PlagueHO, I fixed the merge conflict, renamed the variables and changed some test output. I guess I forgot to change it when I created the file. 👍

PlagueHO commented:
Really great job @Warryz!

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xnetworking/289#-:-L03ZtPKrgAFA2MStlTw:bainyq0)*
<!-- Sent from Reviewable.io -->
